examples/wannier.jl

```diff
@@ -0,0 +1,144 @@
+# # Wannierization using Wannier.jl or Wannier90
+#
+# DFTK features an interface with the programs
+# [Wannier.jl](https://wannierjl.org) and [Wannier90](http://www.wannier.org/),
+# in order to compute maximally-localized Wannier functions (MLWFs)
+# from an initial self consistent field calculation.
+# All processes are handled by calling the routine `wannier_model` (for Wannier.jl) or `run_wannier90` (for Wannier90).
+#
+# !!! warning "No guarantees on Wannier interface"
+#     This code is at an early stage and has so far not been fully tested.
+#     Bugs are likely and we welcome issues in case you find any!
+#
+# This example shows how to obtain the MLWFs corresponding
+# to the first five bands of graphene. Since the bands 2 to 11 are entangled,
+# 15 bands are first computed to obtain 5 MLWFs by a disantanglement procedure.
+
+using DFTK
+using Plots
+using Unitful
+using UnitfulAtomic
+
+d = 10u"Å"
+a = 2.641u"Å"  # Graphene Lattice constant
+lattice = [a  -a/2    0;
+           0  √3*a/2  0;
+           0     0    d]
+
+C = ElementPsp(:C; psp=load_psp("hgh/pbe/c-q4"))
+atoms     = [C, C]
+positions = [[0.0, 0.0, 0.0], [1//3, 2//3, 0.0]]
+model  = model_PBE(lattice, atoms, positions)
+basis  = PlaneWaveBasis(model; Ecut=15, kgrid=[5, 5, 1])
+nbandsalg = AdaptiveBands(basis.model; n_bands_converge=15)
+scfres = self_consistent_field(basis; nbandsalg, tol=1e-5);
+
+# Plot bandstructure of the system
+
+plot_bandstructure(scfres; kline_density=10)
+
+# ## Wannierization with Wannier.jl
+#
+# Now we use the `wannier_model` routine to generate a Wannier.jl model
+# that can be used to perform the wannierization procedure.
+# For now, this model generation produces file in the Wannier90 convention,
+# where all files are named with the same prefix and only differ by
+# their extensions. By default all generated input and output files are stored
+# in the subfolder "wannierjl" under the prefix "wannier" (i.e. "wannierjl/wannier.win",
+# "wannierjl/wannier.wout", etc.). A different file prefix can be given with the
+# keyword argument `fileprefix` as shown below.
+#
+# We now produce a simple Wannier model for 5 MLFWs.
+#
+# For a good MLWF, we need to provide initial projections that resemble the expected shape
+# of the Wannier functions.
+# Here we will use:
+# - 3 bond-centered 2s hydrogenic orbitals for the expected σ bonds
+# - 2 atom-centered 2pz hydrogenic orbitals for the expected π bands
+
+using Wannier # Needed to make wannier_model available
+
+## Helper functions to produce hydrogenic starting guesses
+s_guess(center) = DFTK.HydrogenicWannierProjection(center, 2, 0, 0, C.Z)
+pz_guess(center) = DFTK.HydrogenicWannierProjection(center, 2, 1, 0, C.Z)
+
+wannier_model = wannier_model(scfres;
```

Comment:
this won't work as you cannot use the name of a function as a variable

Reply:
`wannier_model` is a very natural name for the variable, so I chose to go for a `Wannier.Model` method to fix this issue.